### PR TITLE
Remove work around for single threaded COMGR

### PR DESCRIPTION
**Pull request by JehandadKhan**
Removing in light of [SWDEV-322000](https://ontrack-internal.amd.com/browse/SWDEV-322000)

### CODE REVIEW

**Comment by JehandadKhan** [comment]
FYI @atamazov

**Comment by junliume** [comment]
@JehandadKhan  CI is complaining:
```
[2023-02-10T19:29:02.481Z] clang (LLVM option parsing)clang (LLVM option parsing): CommandLine Error: Option 'h' registered more than once!

[2023-02-10T19:29:02.481Z] LLVM ERROR: inconsistency in registered CommandLine options

[2023-02-10T19:29:02.481Z] : CommandLine Error: Option 'h' registered more than once!

[2023-02-10T19:29:02.481Z] LLVM ERROR: inconsistency in registered CommandLine options

[2023-02-10T19:29:02.481Z] Aborted (core dumped)

File: src/hipoc/hipoc_program.cpp [hunk 1]
```diff
@@ -62,10 +62,6 @@ MIOPEN_DECLARE_ENV_VAR(MIOPEN_DEBUG_USE_HIPRTC)
 
 #define MIOPEN_WORKAROUND_ISSUE_1359 1
 
-#if MIOPEN_USE_COMGR
-#define MIOPEN_WORKAROUND_ROCM_COMPILER_SUPPORT_ISSUE_27 1
```
Comment (atamazov) [inline src/hipoc/hipoc_program.cpp:66]: We need to know the HIP version from which the defect was fixed and define this macro to 0 from that version. HIP version can be taken from the hip-clang mainline where the defect is fixed (it should be reported in SWDEV).
```C++
#define MIOPEN_WORKAROUND_ROCM_COMPILER_SUPPORT_ISSUE_27 (HIP_PACKAGE_VERSION_FLAT < xyyyzzzzzzULL)
```
This should resolve https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1977#issuecomment-1426844139

See #1685 for more info about W/A lifecycle, thanks.

**Review by atamazov — CHANGES_REQUESTED** [review]
Please maintain backward compatibility.

File: src/hipoc/hipoc_program.cpp (hunk 1 above)
Comment (JehandadKhan) [reply src/hipoc/hipoc_program.cpp:66]: Will do

**Comment by JehandadKhan** [comment]
Closing this PR since @atamazov  has a full blown fix in progress.

**Pull request closed without merging**